qiskit/primitives/containers/bindings_array.py

```diff
@@ -162,12 +132,7 @@ def kwvals(self) -> dict[tuple[str, ...], np.ndarray]:
     @property
     def num_parameters(self) -> int:
         """The total number of parameters."""
-        return sum(val.shape[-1] for val in chain(self.vals, self._kwvals.values()))
-
-    @property
-    def vals(self) -> list[np.ndarray]:
-        """The non-keyword values of this array."""
-        return self._vals
+        return sum(val.shape[-1] for val in self._kwvals.values())
```

Comment:
Since this class is nominally immutable, we could cache the num parameters on the first call

Reply:
Done in c654f1c